numpy/ma/core.py

```diff
@@ -219,7 +219,7 @@ def default_fill_value(obj):
         defval = default_filler['i']
     elif isinstance(obj, str):
         defval = default_filler['S']
-    elif isinstance(obj, unicode):
+    elif sys.version_info[0] < 3 and isinstance(obj, unicode):
```

Comment:
Maybe we should use compatible filler values also, `sixu('N/A')` for unicode and `b'N/A'` for bytes. Not sure about the usefulness of 'N/A' however.